Binance Spot adapter missing listenKeyExpired event handling

# Bug Report

## Confirmation

- [x] I've re-read the relevant sections of the documentation.
- [x] I've searched existing issues and discussions to avoid duplicates.
- [x] I've reviewed or skimmed the source code (or examples) to confirm the behavior is not by design.
- [x] I've tested this issue using a recent *development* wheel and can still reproduce it.

### Expected Behavior

When Binance sends a `listenKeyExpired` WebSocket event on the Spot user data stream, it should be handled gracefully with a warning log (similar to the Futures adapter).

### Actual Behavior

The Binance Spot adapter throws a `ValidationError` because `listenKeyExpired` is not defined in the `BinanceSpotEventType` enum:

```
[ERROR] BINANCE-TEST-001.ExecClient-BINANCE: Error on handling b'{"stream":"...","data":{"e":"listenKeyExpired","E":1767502801330,"listenKey":"..."}}'
ValidationError(Invalid enum value 'listenKeyExpired' - at `$.data.e`)
Traceback (most recent call last):
  File ".../nautilus_trader/adapters/binance/spot/execution.py", line 241, in _handle_user_ws_message
    wrapper = self._decoder_spot_user_msg_wrapper.decode(raw)
```

### Steps to Reproduce the Problem

1. Run a Binance Spot live trading node with user data stream subscribed
2. Wait for HTTP issues that cause listen key ping failures (common on testnet)
3. After 60+ minutes without successful ping, Binance sends `listenKeyExpired` event
4. The adapter throws a ValidationError instead of handling gracefully

### Code Snippets or Logs

The Futures adapter already handles this event:

```python
# nautilus_trader/adapters/binance/futures/enums.py
class BinanceFuturesEventType(Enum):
    ...
    LISTEN_KEY_EXPIRED = "listenKeyExpired"

# nautilus_trader/adapters/binance/futures/execution.py
self._futures_user_ws_handlers = {
    ...
    BinanceFuturesEventType.LISTEN_KEY_EXPIRED: self._handle_listen_key_expired,
}

def _handle_listen_key_expired(self, raw: bytes) -> None:
    self._log.warning("Listen key expired")
```

But the Spot adapter is missing this entirely.

### Specifications

- OS platform: macOS (Darwin 22.3.0)
- Python version: 3.12
- `nautilus_trader` version: develop branch (editable install)

--- a/nautilus_trader/adapters/binance/spot/enums.py
+++ b/nautilus_trader/adapters/binance/spot/enums.py
@@ -99,6 +99,7 @@ class BinanceSpotEventType(Enum):
     balanceUpdate = "balanceUpdate"
     executionReport = "executionReport"
     listStatus = "listStatus"
+    listenKeyExpired = "listenKeyExpired"
 
 
 class BinanceSpotEnumParser(BinanceEnumParser):

--- a/nautilus_trader/adapters/binance/spot/execution.py
+++ b/nautilus_trader/adapters/binance/spot/execution.py
@@ -124,6 +124,7 @@ class BinanceSpotExecutionClient(BinanceCommonExecutionClient):
             BinanceSpotEventType.executionReport: self._handle_execution_report,
             BinanceSpotEventType.listStatus: self._handle_list_status,
             BinanceSpotEventType.balanceUpdate: self._handle_balance_update,
+            BinanceSpotEventType.listenKeyExpired: self._handle_listen_key_expired,
         }
 
         # Websocket spot schema decoders
@@ -259,3 +260,6 @@ class BinanceSpotExecutionClient(BinanceCommonExecutionClient):
 
     def _handle_balance_update(self, raw: bytes) -> None:
         self.create_task(self._update_account_state())
+
+    def _handle_listen_key_expired(self, raw: bytes) -> None:
+        self._log.warning("Listen key expired")

